hppc/src/main/templates/com/carrotsearch/hppc/KTypeVTypeHashMap.java

```diff
@@ -533,6 +533,25 @@ public void indexInsert(int index, KType key, VType value) {
     }
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public VType indexRemove(int index) {
+    assert index >= 0 : "The index must point at an existing key.";
+    assert index <= mask ||
+            (index == mask + 1 && hasEmptyKey);
+
+    VType previousValue = Intrinsics.<VType> cast(values[index]);
```

Comment:
this is perhaps being overly picky but previous value for the empty key could be just returned without accessing the array... on the other hand - maybe it's good that it doesn't cheat and returns what really was in that slot? Can't decide.

Reply:
I don't see how we can get the previous value without accessing the array. Could you explain me?
The remove() method gets the empty key's previous value from the array.

Reply:
Duh. I'm sorry, I don't think I was thinking clearly. The code is obviously correct.